File: apps/cargo-tracker/src/CargoTracker.Web/Shipper/Application/Internal/CommandServices/RegisterShipperCommand.cs
namespace CargoTracker.Shipper.Application.Internal.CommandServices;

/// <summary>荷主登録コマンド（US02 個人 / US03 法人）。法人のみ契約番号・割引率を持つ。</summary>
public sealed record RegisterShipperCommand(
    bool IsCorporate,
    string Name,
    string Email,
    string? Phone,
    string? Address,
    string? ContractNumber,
    decimal? DiscountRate);

File: apps/backend/bookingms/src/main/java/com/example/bookingms/application/internal/RegisterShipperCommand.java
package com.example.bookingms.application.internal;

import com.example.bookingms.domain.model.ContractNumber;
import com.example.bookingms.domain.model.DiscountRate;
import com.example.bookingms.domain.model.ShipperType;

public record RegisterShipperCommand(
        ShipperType type, String name, String email, String address, String phone,
        ContractNumber contractNumber, DiscountRate discountRate) {

    /** 契約情報を伴わない登録。 */
    public RegisterShipperCommand(
            ShipperType type, String name, String email, String address, String phone) {
        this(type, name, email, address, phone, null, null);
    }
}
